src/pages/Vaults/Vault/components/TransactionHistory/TransactionTable.tsx

```diff
@@ -31,18 +38,46 @@ const RequestCell = ({ request, date }: any) => (
   </StyledRequestCell>
 );
 
-const _TransactionTable = ({ data, ...props }: TransactionTableProps): JSX.Element => {
-  const rows = data.map(({ request, amount, date, status }, key) => ({
+const TransactionTable = ({ data, ...props }: TransactionTableProps): JSX.Element => {
+  const [selectedTableRow, setSelectedTableRow] = useState<any>(undefined);
+
+  // TODO: add type here
+  const rows = data.map(({ request, amount, requestData, date, status }, key) => ({
     id: key,
     amount,
+    requestData,
+    type: request,
     request: <RequestCell request={request} date={date} />,
     status: <TransactionStatusTag status={status} />
   }));
 
-  return <StyledTable columns={columns} rows={rows} {...props} />;
+  return (
+    <>
+      <StyledTable
+        onRowAction={(key) => setSelectedTableRow(rows.find((row) => row.id === key))}
+        columns={columns}
+        rows={rows}
+        {...props}
+      />
+      {/* TODO: these modals should be refactored/replaced */}
+      {selectedTableRow?.type === 'Issue' && (
+        <IssueRequestModal
+          open={selectedTableRow.type === 'Issue'}
+          onClose={() => setSelectedTableRow(undefined)}
+          request={selectedTableRow.requestData}
+        />
+      )}
+      {selectedTableRow?.type === 'Redeem' && (
+        <RedeemRequestModal
+          open={selectedTableRow.type === 'Redeem'}
+          onClose={() => setSelectedTableRow(undefined)}
+          request={selectedTableRow.requestData}
+        />
+      )}
+      ;
+    </>
+  );
 };
 
-const TransactionTable = memo(_TransactionTable);
```

Comment:
Removed memo here as I've introduced an instance of useState to handle the modal